### PR TITLE
py sol for bracket seq II 

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

AC: https://cses.fi/paste/95b7d3353fdea5c6a57dad/

### CODE REVIEW

File: content/6_Advanced/Catalan.mdx [hunk 1]
```diff
@@ -186,7 +186,9 @@ def inverses():
 		inv[i - 1] = inv[i] * i % MOD
 
 
-def choose(n: int, r: int):
+def choose(n: int, r: int) -> int:
+	if n - r < 0:
```
Comment (SansPapyrus683) [inline content/6_Advanced/Catalan.mdx:190]: just n < r

File: solutions/advanced/cses-2187.mdx [hunk 2]
```diff
@@ -107,4 +107,82 @@ int main() {
...
+
+```py
+MAXN = 10**6
+MOD = 10**9 + 7
+
+fac = [0] * (MAXN + 1)
+inv = [0] * (MAXN + 1)
+
+
+def exp(x: int, n: int, m: int) -> int:
```
[the remark this answers is not in the document]
Comment (freakin23) [reply solutions/advanced/cses-2187.mdx:120]: ok

File: content/6_Advanced/Catalan.mdx (hunk 1 above)
Comment (freakin23) [reply content/6_Advanced/Catalan.mdx:190]: ok

File: solutions/advanced/cses-2187.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/advanced/cses-2187.mdx:120]: on second thougt this entire codesnip is unnecessary bc python has a builtin comb function

[the remark this answers is not in the document]
Comment (freakin23) [reply solutions/advanced/cses-2187.mdx:120]: from next time i will keep this in mind, for now let's stick to the basics

[the remark this answers is not in the document]
Comment (freakin23) [reply solutions/advanced/cses-2187.mdx:120]: usign comb function result in TLE for 3 TCs.

File: solutions/advanced/cses-2187.mdx [hunk 3]
```diff
@@ -107,4 +107,84 @@ int main() {
...
+def factorial():
+	fac[0] = 1
+	for i in range(1, MAXN + 1):
+		fac[i] = fac[i - 1] * i % MOD
+
+
+def inverses():
+	inv[MAXN] = exp(fac[MAXN], MOD - 2, MOD)
+	for i in range(MAXN, 0, -1):
+		inv[i - 1] = inv[i] * i % MOD
```
Comment (SansPapyrus683) [inline solutions/advanced/cses-2187.mdx:140]: `exp` and `inverses` aren't needed

Comment (freakin23) [reply solutions/advanced/cses-2187.mdx:140]: `inv` is used in `choose` function

Comment (freakin23) [reply solutions/advanced/cses-2187.mdx:140]: i will remove the `exp` function

File: content/6_Advanced/Catalan.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply content/6_Advanced/Catalan.mdx:190]: not changed?

Comment (freakin23) [reply content/6_Advanced/Catalan.mdx]: my bad, didn't paid attention to other file

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**